### PR TITLE
feat(unsubscriber): add reason to unsubscription request

### PR DESCRIPTION
**Pull request by stanleynguyen**
## Problem

Add reason to unsubscribe flow

- [Ticket](https://www.notion.so/opengov/Update-unsub-flow-copy-48f85ed8622142668697703f11a2241c)

## Solution

Update unsub flow to the new one

## Deployment Checklist

_Any tasks that need to be done on the environment before releasing of this PR (e.g. Setting environment variables, running migrations, etc). If there's no task to be done, put "N/A"_

- [x] Migrate database to add the new columns of `unsubscribers` table

### CODE REVIEW

**Comment by tan-zx** [comment]
image
hmm testing on staging and got this

**Comment by tan-zx** [comment]
Some thoughts on the design (thinking out loud here)
- Currently, if the user unsubscribes, changes his mind, and changes his mind again, they will get an "Invalid unsubscribe request"
  - Is this expected behavior or is this user flow sth we might want to support?
  - Why this might happen: user changes their mind and wishes to change the provided reason

image

Seems like there is safeguard against malicious unsubscribing by adding the hash in the submission body

File: backend/src/core/services/unsubscriber.service.ts [hunk 1]
```diff
@@ -44,11 +44,31 @@ const validateHash = ({
 const findOrCreateUnsubscriber = ({
   campaignId,
   recipient,
+  reason,
```
Comment (tan-zx) [inline backend/src/core/services/unsubscriber.service.ts:47]: need to add this to the JSDoc above (or just delete the useless JSDoc grrr)

**Review by tan-zx — APPROVED** [review]
left some comments on the design; otherwise gltm

File: frontend/src/components/unsubscribe/tests/Unsubscribe.test.tsx [hunk 2]
```diff
@@ -62,14 +70,16 @@ test('successfully unsubscribes user from a campaign', async () => {
...
   server.use(...mockApis())
   renderUnsubscribe()
 
+  await unsubscribe()
+
   // Wait for the component to fully load
   const stayButton = await screen.findByRole('button', {
-    name: /stay/i,
+    name: /subscribe me back/i,
   })
```
Comment (tan-zx) [inline frontend/src/components/unsubscribe/tests/Unsubscribe.test.tsx:83]: Do we want to mock the error cases too? E.g. when the hash is wrong, or when the "Other" reason is too long (is there any limit on this?)

Comment (stanleynguyen) [reply frontend/src/components/unsubscribe/tests/Unsubscribe.test.tsx:82]: Nope no restriction on length of Other reason

**Pull request merged**